record-editor.py

```diff
@@ -1691,40 +1721,42 @@ def create_layout( self ):
         editing_layout.addWidget( self.artDateLineEdit,
                                   5, 1 )
 
-
-        editing_box = QGroupBox()
-        editing_box.setLayout( editing_layout )
-
         #   vertical layout of the photo preview and everything else.
         main_layout = QVBoxLayout()
         main_layout.setContentsMargins( 0, 0, 0, 0 )
         main_layout.setSpacing( 0 )
 
         main_layout.addWidget( self.photoPreview )
         main_layout.setAlignment( self.photoPreview, Qt.AlignCenter ) # the preview should be centered.
-        main_layout.addWidget( editing_box )
+        main_layout.addLayout( editing_layout )
+        main_layout.setStretchFactor( self.photoPreview, 1 )
 
         self.centralWidget = QGroupBox()
         self.centralWidget.setLayout( main_layout )
 
+        self.setCentralWidget( self.centralWidget )
+
     def create_menus( self ):
         """
-        Creats the menus for an ArtRecordEditor.
+        Creates the menus for an ArtRecordEditor.
 
         Takes no arguments.
 
         Returns nothing.
         """
-
-        self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
-                                  triggered=self.close )
+        # XXX: for whatever reason, closing the window via menu action
+        # when there are more than two QMainWindow instances causes
+        # a hard crash. For now the user will have to resort to using
+        # the X button, which does not cause the crash.
+        # self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
+        #                           triggered=self.close )
```

Comment:
Okay, now I see what you're talking about - activating the menu nukes it.
Apparently I can't read.

I'm good with leaving this in and then indirectly addressing it by
migrating to modal widgets.  That would fall inline with code I'd like to
commit that pulls in the *RecordEditor classes for a supplemental editor
embedded within another application (in this case a viewer).

Go ahead and rollback this change within this PR.  Do you want to tackle
migrating the inheritance issue in this PR or set it aside for a future
PR?  Either way it shouldn't get in the way of what I'm doing.

On Sun, Aug 21, 2016 at 7:31 PM, swagneto <notifications@github.com> wrote:

> In record-editor.py
> <https://github.com/gthomsen/streetart-analysis/pull/1#discussion_r75607067>
> :
>
> >
> >          Takes no arguments.
> >
> >          Returns nothing.
> >          """
> > -
> > -        self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
> > -                                  triggered=self.close )
> > +        # XXX: for whatever reason, closing the window via menu action
> > +        # when there are more than two QMainWindow instances causes
> > +        # a hard crash. For now the user will have to resort to using
> > +        # the X button, which does not cause the crash.
> > +        # self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
> > +        #                           triggered=self.close )
>
> I'm able to reproduce on your unmodified master branch, but the key thing
> is you have to *click* the Close Window action, not use the shortcut
> (which seems to work just fine). I get this back from pycham after the hard
> crash:
>
> Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
>
> I don't want to upset your workflow so I'm fine rolling back this change
> (can it be done from within this pull request?). Better fix is to probably
> avoid having Photo/ArtRecordEditor inherit QMainWindow and make them modal
> QWidgets instead, but that would be pretty low priority fix I would imagine.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gthomsen/streetart-analysis/pull/1/files/41ae18f71c9958a8c58e39e1eab98c472e99a783#r75607067>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJjQY3OBAHDBOnbJxcRkbivwYQbFTWZUks5qiO32gaJpZM4JmEfF>
> .
>